Storage/SQL/BrowserDB.swift

```diff
@@ -36,48 +36,43 @@ open class BrowserDB {
     // instance has been initialized (schema is created/updated).
     public func touch() -> Success {
         let deferred = Success()
-        var err: NSError? = nil
-        withConnection(&err) { connection, error -> Void in
-            guard let _ = connection as? ConcreteSQLiteDBConnection else {
-                deferred.fill(Maybe(failure: DatabaseError(description: "Could not establish a database connection")))
-                return
-            }
 
-            deferred.fill(Maybe(success: ()))
+        do {
+            try withConnection { connection -> Void in
+                guard let _ = connection as? ConcreteSQLiteDBConnection else {
+                    throw DatabaseError(description: "Could not establish a database connection")
+                }
+
+                deferred.fill(Maybe(success: ()))
+            }
+        } catch let err as NSError {
+            deferred.fill(Maybe(failure: err))
```

Comment:
Careful here. This is only valid if no failable code in `withConnection` itself runs _after_ the block the caller provides, which is not something we can capture with the signature. 

If someone ever adds code like this, it's possible to fill the deferred twice — once inside the block with success, and again here in the error handler.

Add this to the list of problems that Rust makes visible :/